builder/oracle/classic/step_add_keys.go

```diff
@@ -0,0 +1,58 @@
+package classic
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-oracle-terraform/compute"
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/multistep"
+)
+
+type stepAddKeysToAPI struct{}
+
+func (s *stepAddKeysToAPI) Run(state multistep.StateBag) multistep.StepAction {
+	// get variables from state
+	ui := state.Get("ui").(packer.Ui)
+	ui.Say("Adding SSH keys to API...")
+	config := state.Get("config").(*Config)
+	client := state.Get("client").(*compute.ComputeClient)
+
+	// grab packer-generated key from statebag context.
+	sshPublicKey := strings.TrimSpace(state.Get("publicKey").(string))
+
+	// form API call to add key to compute cloud
+	sshKeyName := fmt.Sprintf("/Compute-%s/%s/packer_generated_key", config.IdentityDomain, config.Username)
+
+	sshKeysClient := client.SSHKeys()
+	sshKeysInput := compute.CreateSSHKeyInput{
+		Name:    sshKeyName,
+		Key:     sshPublicKey,
+		Enabled: true,
+	}
+
+	// Load the packer-generated SSH key into the Oracle Compute cloud.
+	keyInfo, err := sshKeysClient.CreateSSHKey(&sshKeysInput)
+	if err != nil {
```

Comment:
I think we should fail if a key exists. I might be missing some context around keys, but in general we should be creating keys with collision-resistant names (i.e. uuids). The other use-case is where the user has a key they want to use, and will pass in the name directly. We should make sure we support both of these if it's technically possible.

Reply:
That's fine. I'll change this to use a UUID instead of a static name.